### PR TITLE
added date to SP, started on position at time formula

### PR DESCRIPTION
**Pull request by clrosenbluth**

### CODE REVIEW

File: src/main/java/PositionAtTime.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        for (String[] record : records)
+        {
+            String t_date = record[0];
+            if (LocalDate.parse(t_date).isAfter(date))
+            {
+                continue;
+            }
+
+            // otherwise:
+            Double quantity = Double.parseDouble(record[2]);
```
Comment (sarayaffe) [inline src/main/java/PositionAtTime.java]: its not record[1] for quantity and record[4] for rate?

File: src/main/java/PositionAtTime.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            Double rate = Double.parseDouble(record[5]);
+            sum += (quantity * rate);
```
Comment (sarayaffe) [inline src/main/java/PositionAtTime.java]: not quantity/rate?...I may be thinking backwards, just confirming.

**Review by sarayaffe — COMMENTED** [review]
Has this been tested at all?

File: src/main/java/PositionAtTime.java (hunk 1 above)
Comment (clrosenbluth) [reply src/main/java/PositionAtTime.java]: With the form of the record on line 20 (I added the date to the database), 0 is date, 2 is quantity, 5 is rate

File: src/main/java/PositionAtTime.java (hunk 2 above)
Comment (clrosenbluth) [reply src/main/java/PositionAtTime.java]: This will depend on how we're treating the rate. Needs to be determined.

**Review by sarayaffe — APPROVED** [review]
@MDreyfuss want to take a look to understand whats going on before Chana merges?

File: src/main/java/PositionAtTime.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+                // otherwise:
+                String type = record[3];
+                String maturityD = record[6];
+                LocalDate maturityLD = maturityD == null || maturityD.equals("") ?
+                        null :
+                        LocalDate.parse(maturityD);
+
+                boolean isSpot = type.equals("spot");
+                boolean isGoodFuture = type.equals("future")
```
Comment (sarayaffe) [inline src/main/java/PositionAtTime.java]: if we get around to it, maybe we can add a display of pending transactions/positions for the forwards that havent matured yet.

**Pull request merged**